src/bootstrap/compile.rs

```diff
@@ -1102,6 +1102,11 @@ impl Step for Assemble {
             let src_exe = exe("lld", target_compiler.host);
             let dst_exe = exe("rust-lld", target_compiler.host);
             builder.copy(&lld_install.join("bin").join(&src_exe), &libdir_bin.join(&dst_exe));
+            // for gcc-lld linker flavor
+            builder.copy(
+                &libdir_bin.join(&dst_exe),
```

Comment:
```suggestion
                &lld_install.join("bin").join(&src_exe),
```
This is a bit nicer IMHO.